library/Respect/Data/Event/EventManager.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace Respect\Data\Event;
+
+use Respect\Data\Event\Interfaces\EventManager as EventManagerInterface;
+use Respect\Data\Event\Interfaces\Listener;
+
+class EventManager implements EventManagerInterface
+{
+    /*
+     * This is the events array
+     * @var array
+     * @example
+     * array(
+     *     "EventName1" => array(
+     *         0 => callable,
+     *         1 => concreteListener
+     *         ...
+     *     )
+     *     ...
+     * )
+     */
+    private $events;
+
+    /**
+     * Construct
+     */
+    public function __construct()
+    {
+        $this->events = array();
+    }
+
+    /**
+     * Add a callable or listener to a new or existent event
+     * @param string $eventName The event to attach the listener
+     * @param \Respect\Data\Event\Interfaces\Listener|\callable $action The event listener callback
+     * @return void
+     */
+    public function on($eventName, $action)
+    {
+        switch (true) {
+        case ( (is_callable($action)) || ($action instanceof Listener) ):
+
+            // The block below you can use for control or remove it
+            $events = $this->getEvents();
+            if (array_key_exists("attach", $events)) {
+                $this->dispatch("attach", array($eventName, $action));
+            }
+
+            $this->events[$eventName][] = $action;
+            break;
+        default:
+            $message = 'Invalid type of action provided for event manager';
+            throw new \InvalidArgumentException($message, 400);
+        }
```

Comment:
Probably for future implementations or checks. I don't see any trouble by changing it.
@Rafael-BP could you?